cmd/cluster/from_infra_id.go

```diff
@@ -79,10 +82,38 @@ func (ops *fromInfraIdOptions) run(cmd *cobra.Command, args []string) error {
 			return renderOutput(cluster, ops.globalOpts.Output)
 		}
 	}
-	_, err = fmt.Fprintf(os.Stderr, "No clusters found matching %s\n", infraId)
+	mostRecentMatchingSub, err := getLatestMatchingSubscription(ocmClient, clusterName)
+	if err != nil {
+		_, _ = fmt.Fprintf(os.Stderr, "No clusters or subscriptions found matching %s\n", infraId)
+	}
+	_, _ = fmt.Fprintf(
+		os.Stderr,
+		"No clusters found for %s; a possibly matching subscription %s is %s and was last updated at %s\n",
+		infraId,
+		mostRecentMatchingSub.ID(),
+		mostRecentMatchingSub.Status(),
+		mostRecentMatchingSub.UpdatedAt(),
+	)
 	return err
 }
 
+func getLatestMatchingSubscription(ocmClient *sdk.Connection, clusterName string) (*amv1.Subscription, error) {
+	subResponse, err := ocmClient.AccountsMgmt().V1().Subscriptions().List().Search(fmt.Sprintf("display_name='%s' and managed='true'", clusterName)).Send()
+	if err != nil {
+		return nil, err
+	}
+	if subResponse.Size() == 0 {
+		return nil, fmt.Errorf("no subscriptions found for cluster name %s", clusterName)
+	}
+	subs := subResponse.Items().Slice()
+	sort.SliceStable(subResponse.Items().Slice(), func(i, j int) bool {
+		return subs[i].UpdatedAt().Before(subs[j].UpdatedAt())
```

Comment:
Could we instead change this line to `return subs[i].UpdatedAt().After(subs[j].UpdatedAt())`, and then down on line 113 just grab `subs[0]` ?

Reply:
Huh, you highlighted a line with a bug. I'm not using `subs` in `SliceStable`'s first parameter. I'll make both changes while I'm there.

Reply:
Updated

Reply:
Funny enough I ended up adding `len(subs)` to the output message, but I'm going to leave the sorting to allow us to pull the 0 index.